aten/src/ATen/native/utils/ParamsHash.h

```diff
@@ -38,5 +38,57 @@ struct ParamsEqual {
   }
 };
 
+// Provide explicit byte-for-byte constructors to avoid uwittingly leaving
+// padding bytes unitialized (e.g., when passing Params by value)
+template <typename T>
+struct ParamsWrapper {
+  T pod;
+  static_assert(std::is_standard_layout_v<T>, "ParamsWrapper cannot wrap non-POD data");
+
+  ParamsWrapper() {
+    memset(&(this->pod), 0, sizeof(T));
+  }
+
+  ParamsWrapper(const ParamsWrapper<T> &other) {
+    memcpy(&(this->pod), &(other.pod), sizeof(T));
+  }
+
+  ParamsWrapper(ParamsWrapper<T> &&other) {
+    memcpy(&(this->pod), &(other.pod), sizeof(T));
+  }
+};
```

Comment:
According to the rule of three/five, we should also define `operator=`.
https://en.cppreference.com/w/cpp/language/rule_of_three

Reply:
Makes sense, will add.